Description: Use system libev
Author: Jérémy Lal <kapouer@melix.org>
Forwarded: not-needed
Last-Update: 2012-09-30

--- a/src/node_io_watcher.h
+++ b/src/node_io_watcher.h
@@ -23,7 +23,7 @@
 #define NODE_IO_H_
 
 #include "node_object_wrap.h"
-#include "uv-private/ev.h"
+#include <ev.h>
 
 namespace node {
 
--- a/deps/uv/include/uv-private/uv-unix.h
+++ b/deps/uv/include/uv-private/uv-unix.h
@@ -24,7 +24,7 @@
 
 #include "ngx-queue.h"
 
-#include "ev.h"
+#include <ev.h>
 #include "eio.h"
 
 #include <sys/types.h>
--- a/deps/uv/config-unix.mk
+++ b/deps/uv/config-unix.mk
@@ -21,8 +21,8 @@
 E=
 CSTDFLAG=--std=c89 -pedantic -Wall -Wextra -Wno-unused-parameter
 CFLAGS += -g
-CPPFLAGS += -Isrc -Isrc/unix/ev
-LINKFLAGS=-lm
+CPPFLAGS += -Isrc
+LINKFLAGS=-lm 
 
 CPPFLAGS += -D_LARGEFILE_SOURCE
 CPPFLAGS += -D_FILE_OFFSET_BITS=64
@@ -66,7 +66,7 @@
 EIO_CONFIG=config_linux.h
 CSTDFLAG += -D_GNU_SOURCE
 CPPFLAGS += -I/usr/include
-LINKFLAGS+=-ldl -lrt -lcares
+LINKFLAGS+=-ldl -lrt -lcares -lev
 OBJS += src/unix/linux/linux-core.o \
         src/unix/linux/inotify.o    \
         src/unix/linux/syscalls.o
@@ -131,7 +131,7 @@
 RUNNER_LIBS=
 RUNNER_SRC=test/runner-unix.c
 
-uv.a: $(OBJS) src/cares.o src/fs-poll.o src/uv-common.o src/unix/ev/ev.o src/unix/uv-eio.o src/unix/eio/eio.o
+uv.a: $(OBJS) src/cares.o src/fs-poll.o src/uv-common.o src/unix/uv-eio.o src/unix/eio/eio.o
 	$(AR) rcs uv.a $^
 
 src/%.o: src/%.c include/uv.h include/uv-private/uv-unix.h
@@ -140,9 +140,6 @@
 src/unix/%.o: src/unix/%.c include/uv.h include/uv-private/uv-unix.h src/unix/internal.h
 	$(CC) $(CSTDFLAG) $(CPPFLAGS) $(CFLAGS) -c $< -o $@
 
-src/unix/ev/ev.o: src/unix/ev/ev.c
-	$(CC) $(CPPFLAGS) $(CFLAGS) -c src/unix/ev/ev.c -o src/unix/ev/ev.o -DEV_CONFIG_H=\"$(EV_CONFIG)\"
-
 
 EIO_CPPFLAGS += $(CPPFLAGS)
 EIO_CPPFLAGS += -DEIO_CONFIG_H=\"$(EIO_CONFIG)\"
--- a/src/node_signal_watcher.h
+++ b/src/node_signal_watcher.h
@@ -24,7 +24,7 @@
 
 #include "node.h"
 #include "v8.h"
-#include "uv-private/ev.h"
+#include <ev.h>
 
 namespace node {
 
--- a/deps/uv/uv.gyp
+++ b/deps/uv/uv.gyp
@@ -110,7 +110,6 @@
           ],
           'sources': [
             'include/uv-private/eio.h',
-            'include/uv-private/ev.h',
             'include/uv-private/uv-unix.h',
             'src/unix/async.c',
             'src/unix/core.c',
@@ -119,10 +118,6 @@
             'src/unix/eio/eio.c',
             'src/unix/eio/xthread.h',
             'src/unix/error.c',
-            'src/unix/ev/ev.c',
-            'src/unix/ev/ev_vars.h',
-            'src/unix/ev/ev_wrap.h',
-            'src/unix/ev/event.h',
             'src/unix/fs.c',
             'src/unix/internal.h',
             'src/unix/loop.c',
@@ -139,7 +134,7 @@
             'src/unix/uv-eio.c',
             'src/unix/uv-eio.h',
           ],
-          'include_dirs': [ 'src/unix/ev', ],
+          'include_dirs': [],
           'link_settings': {
             'libraries': [ '-lm' ],
             'conditions': [
@@ -178,7 +173,7 @@
             'EIO_CONFIG_H="config_linux.h"',
           ],
           'link_settings': {
-            'libraries': [ '-ldl', '-lrt', '-lcares' ],
+            'libraries': [ '-ldl', '-lrt', '-lcares', '-lev' ],
           },
         }],
         [ 'OS=="solaris"', {
--- a/tools/install.py
+++ b/tools/install.py
@@ -189,7 +189,6 @@
           'src/node_version.h'],
           'include/node/')
   action(['deps/uv/include/uv-private/eio.h',
-          'deps/uv/include/uv-private/ev.h',
           'deps/uv/include/uv-private/ngx-queue.h',
           'deps/uv/include/uv-private/tree.h',
           'deps/uv/include/uv-private/uv-unix.h',
