diff --git a/debian/changelog b/debian/changelog
index 6519fa0..377427f 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,7 @@
+diamond (3.1.1) unstable; urgency=low
+  * local patch to limit network collector statistics; not for upstream
+ -- John Hopkins <jhopkins@mozilla.com>  Fri, 28 Feb 2014 10:38:00 -0500
+
 diamond (3.1.0) unstable; urgency=low
   * new upstream
  -- Rob Smith <kormoc@gmail.com>  Tue, 05 Nov 2012 15:19:00 -0800
diff --git a/src/collectors/network/network.py b/src/collectors/network/network.py
index b282747..addbf19 100644
--- a/src/collectors/network/network.py
+++ b/src/collectors/network/network.py
@@ -109,6 +109,8 @@ class NetworkCollector(diamond.collector.Collector):
         for device in results:
             stats = results[device]
             for s, v in stats.items():
+                if s not in ['rx_bytes', 'tx_bytes']:
+                    continue
                 # Get Metric Name
                 metric_name = '.'.join([device, s])
                 # Get Metric Value
